modules/sov-ibc-transfer/src/context.rs

```diff
@@ -164,7 +158,7 @@ where
                 })?
         };
 
-        if token_id != expected_token_id {
+        if sov_memo.token_id() != expected_token_id {
```

Comment:
the way I see, the memo is not really necessary. but it acts as a double guard making sure the user knows what `token_id` is being used. The transfer app always maintains the unique `token_id`s corresponding to the ibc denom trace.